src/Websites/Websites/Utilities/WebsitesClient.cs

```diff
@@ -97,11 +97,12 @@ public HostingEnvironmentProfile CreateHostingEnvironmentProfile(string resource
             return CmdletHelpers.CreateHostingEnvironmentProfile(WrappedWebsitesClient.SubscriptionId, resourceGroupName, aseResourceGroupName, aseName);
         }
 
-        public void UpdateWebApp(string resourceGroupName, string location, string webAppName, string slotName, string appServicePlan, Site siteEnvelope =null)
+        public void UpdateWebApp(string resourceGroupName, string location, string webAppName, string slotName, string appServicePlan, Site siteEnvelope =null,string appServicePlanRg=null)
         {
             var webSiteToUpdate = new Site()
             {
-                ServerFarmId = appServicePlan,
+                //Condition added to send full qualified ServerFarmId if ASP and WebApp are under different  RGs.
+                ServerFarmId = (string.IsNullOrEmpty(appServicePlanRg)&&resourceGroupName != appServicePlanRg) ? appServicePlan : siteEnvelope.ServerFarmId,
```

Comment:
These seem like they need some test coverage to prevent regressions

Reply:
@Kotasudhakarreddy incase this is already done - resolve this.